aurora/src/main/java/src/config/ChannelManager.java

```diff
@@ -30,17 +31,23 @@
 
     public Optional<ManagedChannel> getChannelByKey(String key) {
         LOGGER.debug("Getting channel with key {}", key);
-        return Optional.ofNullable(channels.get(key));
+        Optional<ManagedChannel> managedChannel = Optional.ofNullable(channels.get(key));
+        if (managedChannel.isPresent() && managedChannel.get()
+                .getState(true).equals(ConnectivityState.READY)) {
+            managedChannel = Optional.ofNullable(null);
```

Comment:
why is managedChannel set to optional of null

Reply:
Fixed